src/pocketmine/updater/AutoUpdater.php

```diff
@@ -154,7 +159,19 @@ protected function checkUpdate(){
 		}else{
 			$this->hasUpdate = false;
 		}
+		return true;
+	}
 
+	/**
+	 * Checks the given response is valid.
+	 */
+	protected function isValidResponse(Array $updateInfo){
```

Comment:
This should be done in the UpdateCheckTask along with the other validity checks so that this can be handled cleanly.